### PR TITLE
Add build badges to README file

### PR DESCRIPTION
**Pull request by iamjazzar**

### CODE REVIEW

**Comment by kaidokert** [comment]
Do we want this in the toplevel project README.md file, or perhaps the full matrix should be in a BUILD_STATUS.md or GITHUB_BUILD_STATUS.md or such ?

**Comment by iamjazzar** [comment]
@kaidokert Done

File: BUILD_STATUS.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Build Status
+
+| Workflow  | Main | 23.lts.1+ | 22.lts.1+ | 21.lts.1+ | 20.lts.1+ | 19.lts.1+ | RC11 | COBALT 9 |
+| --------- | ---- | --------- | --------- | --------- | --------- | --------- | ---- | ---------|
+| Lint      | [![lint](https://github.com/youtube/cobalt_sandbox/actions/workflows/lint.yaml/badge.svg?branch=main&event=push)](https://github.com/youtube/cobalt_sandbox/actions/workflows/lint.yaml) | | | | | | | |
```
Comment (andrewsavage1) [inline BUILD_STATUS.md:5]: Can you permalink to workflows filtered in the same way we filter events for the badge?

Comment (andrewsavage1) [reply BUILD_STATUS.md:5]: I believe it is, so let's link to the filtered list

Comment (iamjazzar) [reply BUILD_STATUS.md]: Done

File: BUILD_STATUS.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+| stub      | [![stub](https://github.com/youtube/cobalt_sandbox/actions/workflows/stub.yaml/badge.svg?branch=main&event=push)](https://github.com/youtube/cobalt_sandbox/actions/workflows/stub.yaml?query=event%3Apush+branch%3Amain) | [![stub_23.lts.1+](https://github.com/youtube/cobalt_sandbox/actions/workflows/stub_23.lts.1+.yaml/badge.svg?branch=23.lts.1%2B&event=push)](https://github.com/youtube/cobalt_sandbox/actions/workflows/stub_23.lts.1+.yaml?query=event%3Apush+branch%3A23.lts.1%2B) | | | | | | |
+| Win32     | [![win32](https://github.com/youtube/cobalt_sandbox/actions/workflows/win32.yaml/badge.svg?branch=main&event=push)](https://github.com/youtube/cobalt_sandbox/actions/workflows/win32.yaml?query=event%3Apush+branch%3Amain) | [![win32_23.lts.1+](https://github.com/youtube/cobalt_sandbox/actions/workflows/win32_23.lts.1+.yaml/badge.svg?branch=23.lts.1%2B&event=push)](https://github.com/youtube/cobalt_sandbox/actions/workflows/win32_23.lts.1+.yaml?query=event%3Apush+branch%3A23.lts.1%2B) | | | | | | |
+
+
+# Nightly builds
```
Comment (isarkis) [inline BUILD_STATUS.md:14]: I don't see main branch in Nightly Builds table.

**Review by isarkis — APPROVED** [review]

**Pull request merged**